src/libertem/executor/utils/delayed_unpack.py

```diff
@@ -0,0 +1,132 @@
+# _unpackable_types = (list, tuple, dict)
+_unpackable_types = {list: lambda x: enumerate(x),
+                     tuple: lambda x: enumerate(x),
+                     dict: lambda x: x.items()}
+merge_fns = {list: lambda lis, el, pos: lis.append(el),
+             dict: lambda dic, el, pos: dic.update({pos: el})}
+
+
+class IgnoreClass:
+    pass
+
+
+class StructDescriptor:
+    def __init__(self, cls, *args, **kwargs):
+        self.cls = cls
+        self.args = args
+        self.kwargs = kwargs
+
+    def __repr__(self):
+        return f'{self.__class__.__name__}({self.cls}, {self.args}, {self.kwargs})'
+
+
+def flatten_nested(el, unpackable_types=None, ignore_types=None):
```

Comment:
I tried, but failed to make mypy accept it. Without that I may very well add wrong type annotations. I'd rather leave it as-is. If you think it is important, maybe you could have a look?

Reply:
As a pragmatic approach since these are internal utility functions: What about adding type annotations at the time we want to make changes and think "It would be nice to have type annotations now"?

Reply:
I can have a look at it. If we want to delay adding type annotations, I would prefer if we had docstrings for these. It's mostly about properly documenting these, so the next time they are touched (or code using them is touched!) you don't have to read and understand the whole function bodies, as often type annotations really clarify the details enough. A docstring would serve the same purpose, even though it is not machine understandable.

Reply:
@matbryan52 maybe you could add some basic docstrings, then? Both @sk1p and I would have to reverse-engineer this code.

Reply:
Looks like this one is now resolved in 3731ee72f4dca5dfaf76fef0e2a054579e4f645a c8c0b0d4945fb74dae63a85bf80f33e53f16ba2a 22bccbce9e57ad22a25becd53c9607371949a9fe - thanks @matbryan52!